### PR TITLE
Apply pykdl_ros

### LINKED ISSUE #1119: Simplify kdl_conversions based on tf2

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Merge after https://github.com/tue-robotics/pykdl_ros/pull/2

Fixes #1119

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
The other packages do not fail. I triggered a rerun, but it doesn't show here. See https://dev.azure.com/tue-robotics/tue_robocup/_build/results?buildId=2896&view=results

**Comment by MatthijsBurgh** [comment]
Missed some `projectToFrame` and `extractVectorStamped` functions

**Comment by MatthijsBurgh** [comment]
> Missed some `projectToFrame` and `extractVectorStamped` functions

Resolved

**Comment by PetervDooren** [comment]
After discussion with @MatthijsBurgh  and @LarsJanssenTUe 
To be merged after #1133 is merged with master.

File: robot_skills/src/robot_skills/world_model_ed.py [hunk 1]
```diff
@@ -103,26 +107,29 @@ def get_entities(self, type="", center_point=VectorStamped(), radius=float('inf'
         :param id: ID of entity
         :param ignore_z: Consider only the distance in the X,Y plane for the radius from center_point
         """
+        if not center_point:
```
Comment (PetervDooren) [inline robot_skills/src/robot_skills/world_model_ed.py:110]:
```suggestion
        if center_point is None:
```

File: robot_skills/src/robot_skills/world_model_ed.py [hunk 2]
```diff
@@ -159,6 +166,8 @@ def get_closest_laser_entity(self, type="", center_point=VectorStamped(), radius
         :param ignore_z: Consider only the distance in the X,Y plane for the radius from center_point criterium.
         :return: list of Entity
         """
+        if not center_point:
```
Comment (PetervDooren) [inline robot_skills/src/robot_skills/world_model_ed.py:169]:
```suggestion
        if center_point is None:
```

File: robot_skills/src/robot_skills/world_model_ed.py [hunk 3]
```diff
@@ -297,14 +306,17 @@ def lock_entities(self, lock_ids, unlock_ids):
...
-    def get_closest_possible_person_entity(self, center_point=VectorStamped(), radius=float('inf')):
+    def get_closest_possible_person_entity(self, center_point=None, radius=float('inf')):
         """ Returns the 'possible_human' entity closest to a certain center point.
 
         :param center_point: (VectorStamped) indicating where the human should be close to
         :param radius: (float) radius to look for possible humans
         :return: (Entity) entity (if found), None otherwise
         """
-        assert center_point.frame_id.endswith("map"), "Other frame ids not yet implemented"
+        if not center_point:
```
Comment (PetervDooren) [inline robot_skills/src/robot_skills/world_model_ed.py:316]:
```suggestion
        if center_point is None:
```

File: robot_smach_states/src/robot_smach_states/manipulation/grasp_point_determination.py [hunk 4]
```diff
@@ -177,7 +185,3 @@ def visualize(self, candidates):
             #####
 
         self._marker_array_pub.publish(msg)
-
```
Comment (PetervDooren) [inline robot_smach_states/src/robot_smach_states/manipulation/grasp_point_determination.py:180]: Shouldnt there be one blank line at the end of file

**Review by PetervDooren — CHANGES_REQUESTED** [review]
Some very minor comments. Consider this an approval once they are addressed.

One question: does pykdl ros now require a timestamp for all vectors and frames? Why so?

File: robot_smach_states/src/robot_smach_states/manipulation/grasp_point_determination.py (hunk 4 above)
Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/manipulation/grasp_point_determination.py:180]: Yes there is. Otherwise github would have shown that.

**Comment by MatthijsBurgh** [comment]
> One question: does pykdl ros now require a timestamp for all vectors and frames? Why so?

Because this matches the [`tf2_ros`](https://github.com/ros/geometry2/blob/noetic-devel/tf2_ros/src/tf2_ros/buffer_interface.py#L169)

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**